pkg/upgrade/control_plane.go

```diff
@@ -267,8 +291,18 @@ func (u *ControlPlaneUpgrader) Upgrade() error {
 			Name:      replacementName,
 		}
 
-		if err := u.managementClusterClient.Get(context.TODO(), key, &replacement); err != nil {
-			return errors.Wrapf(err, "error getting machine %s", key.String())
+		err := u.managementClusterClient.Get(context.TODO(), key, &replacement)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				// if there is a bad replacement machine in the beginning of the
+				// upgrade it is eventually deleted but the reference of the
+				// machine object is still in the slice of machines so just log
+				// this error and continue.
+				u.log.Error(err, "error getting machine", "machine", key.String())
```

Comment:
This could be also an `Info` message, including some of the information there is in the comment above